src/layers/icon-layer/icon-layer.js

```diff
@@ -104,13 +96,40 @@ export default class IconLayer extends Layer {
   get layerInfoModal() {
     return {
       id: 'iconInfo',
-      template: IconInfoModal,
+      template: this._layerInfoModal,
       modalProps: {
         title: 'How to draw icons'
       }
     };
   }
 
+  async getSvgIcons(callback) {
```

Comment:
remove __callback__ param because it is not used.

Do you need this method to be part of this class? I think it could be an external method to fetch all data and use the callback to set the necessary property by providing a callback or the consumer can use await and retrieve the result.

if you initialize more icon layers you may perform the same network calls multiple times. If you externalize the method you may also think about storing the data into a variable /cache to speed subsequent calls

Reply:
I have thought about it before, and decided to go this route. The svg icons are only used in the icon layer. So the loading part should be handled in the layer life cycle. This will make it easier later when we modulize kepler.gl,  packaging the layers of kepler.gl into its own package nad and have users to selectively import layers they needed. 

I agree if multiple icon layer were created, this call will be called multiple times. But this is not always a bad thing. Icon layer should be able to accept icon props and draw difference selection of svg icons. So you can image we will add a icons prop to this layer, and different layers might how a different collection of icons